public/index.js

```diff
@@ -101,26 +102,40 @@ function GenerateArrowIcons()
     return icons;
 }
 
-async function DisplayRoute(routeId, color)
+async function DisplayRoute(routeId, color, routeName)
 {
-    displayedRoutes.forEach(route => route.setMap(null));
-
-    let response = await fetch('shape-coords-for-route-id?' + new URLSearchParams({ routeId: routeId }));
+    const btns = document.getElementsByClassName(routeName);
     let coords = [];
-    if (response.ok)
+
+    if (btns[0].style.border != '5px solid rgb(64, 64, 64)')

```

Comment:
I don't love the string hardcoded string comparison. Could we add this to the style.css file and add/remove a "selected" class on the button click instead?